### PR TITLE
Support enumeration for pybind11

### PR DESCRIPTION
**Pull request by jslee02**
Caveat: nested enumeration (probably class) doesn't work with current pybind11 templates.

### CODE REVIEW

File: test/04_enumeration/enumeration.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import unittest
+
+try:
+    import enumeration_pybind11.nested_namespace as py11
+    has_pybind11 = True
+except:
+    has_pybind11 = False
+
+import enumeration_boost_python.nested_namespace as boost
```
Comment (psigen) [inline test/04_enumeration/enumeration.py]: I think I missed this in the other PR.  Where do the `enumeration_xx_yy` namespaces come from here?

**Comment by psigen** [comment]
In general, I am fine with adding more functionality to these pybind11 templates even if they are not complete, as long as we do not introduce regressions.  Each feature makes chimera more useful!

File: test/04_enumeration/enumeration.py (hunk 1 above)
Comment (jslee02) [reply test/04_enumeration/enumeration.py]: That's the name of the binding, which is specified at [here](https://github.com/personalrobotics/chimera/blob/py11_enum/test/04_enumeration/CMakeLists.txt#L3) with suffixes depending on the binding library (`_boost_python` or `_pybind`).

**Comment by jslee02** [comment]
I started to switch the `dartpy` bindings to pybind and am adding necessary features to here whenever I find something missing. I finished `dart::common` and started `dart::math`. Once `dartpy` completely migrate to pybind11 with full API of dart, I'm pretty sure, pybind11 support of chimera will be reasonably sufficient.

**Pull request merged**